modules/Files/endpoints.bal

```diff
@@ -0,0 +1,471 @@
+        file:MetaData fileMetaData = check file:getMetaData(localFilePath);
+        int fileSizeInByte = fileMetaData.size;
+        var createFileResponse = self->createFile(fileShareName, azureFileName, fileSizeInByte, azureFilePath);
+        if (createFileResponse == true) {
+            var uploadResult = putRangeInternal(self.httpClient, fileShareName, localFilePath, azureFileName, self.
+            sasToken, fileSizeInByte, azureFilePath);
+            return uploadResult;
+        } else {
+            return createFileResponse;
+        }
+
+    }
+}
+
+function createFileInternal(http:Client httpClient, string fileShareName, string fileName, 
+                            int  fileSizeInByte, string sasToken, string? azureDirectoryPath = ()) returns @tainted boolean|error {
+    string requestPath  = SLASH + fileShareName;
+    requestPath = azureDirectoryPath is () ? requestPath : (requestPath + SLASH + azureDirectoryPath);
+    requestPath = requestPath + SLASH + fileName + QUESTION_MARK + sasToken;
+    http:Request request = new;
+    map<string> requiredSpecificHeaderes ={
+        "x-ms-file-permission" :"inherit",
+        "x-ms-file-attributes" : "None",
+        "x-ms-file-creation-time" : "now",
+        "x-ms-file-last-write-time" : "now",
+        "Content-Length" : "0",
+        "x-ms-content-length" : fileSizeInByte.toString(),
+        "x-ms-type" : "file"
+    };
+    setSpecficRequestHeaders(request, requiredSpecificHeaderes);
+
+    http:Response response = <http:Response>check httpClient->put(requestPath, request);
+    if (response.statusCode == CREATED) {
+        return true;
+    } else {
+        fail error(getErrorMessage(response));
+    }
+}
+
+function putRangeInternal(http:Client httpClient, string fileShareName, string localFilePath, 
+                        string azureFileName, string sasToken, int fileSizeInByte, string? azureDirectoryPath = ()) returns @tainted boolean|error {
+
+    string requestPath = SLASH + fileShareName;
+    requestPath = azureDirectoryPath is () ? requestPath : (requestPath + SLASH + azureDirectoryPath);
+    requestPath = requestPath + SLASH + azureFileName + QUESTION_MARK + PUT_RANGE_PATH + AMPERSAND + sasToken;
+    stream<io:Block>|io:Error fileStream = io:fileReadBlocksAsStream(localFilePath, 4194304);
+    int index = 0;
+    int remainingBytesAmount = fileSizeInByte;
+    boolean updateStatusFlag = false;
+    if (fileStream is stream<io:Block>) {
+        error? e = fileStream.forEach(
+            function(io:Block byteBlock) {
+                if (remainingBytesAmount > MAX_UPLOADING_BYTE_SIZE) {
+                    http:Request request = new;
+                    map<string> requiredSpecificHeaderes = {
+                        "x-ms-range" : "bytes=" + index.toString() + "-" + (index + MAX_UPLOADING_BYTE_SIZE - 1).toString(),
+                        "Content-Length" : MAX_UPLOADING_BYTE_SIZE.toString(),
+                        "x-ms-write" : "update"
```

Comment:
Since these header values are used in many places, can have constants for them

Reply:
I tried it also.  When we  set as a constant  like this ` X_MS_FILE_PERMISSION:`, ballerina could not identify it. ```An HTTP header that's mandatory for this request is not specified```